plugin/build.gradle.kts

```diff
@@ -243,10 +231,85 @@ java {
   withJavadocJar()
 }
 
-tasks.shadowJar.configure {
-  archiveClassifier.set("")
+if(project.properties["rfg.skipJavadoc"].toString().toBoolean()) {
+  tasks.named("javadoc").configure { enabled = false }
+}
+
+val depsShadowJar = tasks.register<ShadowJar>("depsShadowJar") {
+  archiveClassifier.set("deps")
+  archiveVersion.set("0.0") // constant version to prevent task from rerunning when project version changes
   isEnableRelocation = true
   relocationPrefix = "com.gtnewhorizons.retrofuturagradle.shadow"
+  configurations.add(project.configurations.runtimeClasspath.get())
+  dependencies {
+    // we're already shading this in combinedShadowJar
+    exclude(project(":oldasmwrapper"))
+  }
+}
+
+val mainShadowJar = tasks.register<ShadowJar>("mainShadowJar") {
+  archiveClassifier.set("mainShadow")
+
+  val prefix = "com.gtnewhorizons.retrofuturagradle.shadow"
+
+  from(sourceSets.main.get().output)
+
+  // Adapted from Shadow's RelocationUtil.groovy
+  // We want to relocate references to dependencies but not include the dependencies themselves in this jar.
+  // We also don't want to double-shade RFG classes
+  val packages = mutableSetOf<String>()
+
+  val configurations = listOf(project.configurations.runtimeClasspath.get())
+
+  configurations.iterator().forEach { configuration ->
+    configuration.files.filter {
+      // we're already shading this in combinedShadowJar
+      f -> f != project(":oldasmwrapper").tasks.named<Jar>("allJar").get().archiveFile.get().asFile
+    }.forEach { jar ->
+      JarFile(jar).use {
+        it.entries().iterator().forEach { entry ->
+          if (entry.name.endsWith(".class") && entry.name != "module-info.class") {
+            val pkg = entry.name.substring(0, entry.name.lastIndexOf('/') - 1).replace('/', '.')
+            if (!pkg.startsWith("com.gtnewhorizons.retrofuturagradle")) {
+              packages.add(pkg)
+            }
+          }
+        }
+      }
+    }
+  }
+  packages.iterator().forEach {
+    relocate(it, "${prefix}.${it}")
+  }
+}
+
+val combinedShadowJar = tasks.register<Jar>("combinedShadowJar") {
```

Comment:
Can't you just configure the default tasks.jar instead of making a new one?

Reply:
The problem is the Java plugin pre-configures the `jar` task to copy the main sourceset's output, and I don't know if it's possible to easily cancel that. So I created a new task to let me start from scratch.

Reply:
You can use setFrom instead of from to reset the CopySpec entirely iirc

Reply:
Nevermind, this is not an api on CopySpec

Reply:
A different solution would be to use `eachFile` to filter out the main sourceset's files, but that still seems overly convoluted.

With that said, I could get rid of the `shadowJar` task at least and have `jar` depend on `combinedShadowJar` directly.